tutor/commands/plugins.py

```diff
@@ -105,48 +171,279 @@ def disable(context: Context, plugin_names: list[str]) -> None:
+            )
+
+
+def install_single_file_plugin(location: str) -> None:
+    """
+    Download or copy a single file to the plugins root.
+    """
+    plugin_path = os.path.join(PLUGINS_ROOT, os.path.basename(location))
+    if not plugin_path.endswith(".yml") and not plugin_path.endswith(".py"):
+        plugin_path += ".py"
+    # Read url
+    fmt.echo_info(f"Downloading plugin from {location}...")
+    content = utils.read_url(location)
     # Save file
-    if not os.path.exists(PLUGINS_ROOT):
-        os.makedirs(PLUGINS_ROOT)
+    utils.ensure_file_directory_exists(plugin_path)
     with open(plugin_path, "w", newline="\n", encoding="utf-8") as f:
         f.write(content)
     fmt.echo_info(f"Plugin installed at {plugin_path}")
 
 
+@click.command()
+@click.argument("pattern", default="")
+def search(pattern: str) -> None:
+    """
+    Search in plugin descriptions.
+    """
+    results: list[tuple[str, ...]] = [("NAME", "STATUS", "DESCRIPTION")]
+    for plugin in indexes.iter_cache_entries():
+        if plugin.match(pattern):
+            results.append(
+                (
+                    plugin.name,
+                    plugin_status(plugin.name),
+                    plugin.short_description,
+                )
+            )
+    print(format_table(results))
+
+
+@click.command()
+@click.argument("name", type=IndexPluginName())
+def show(name: str) -> None:
+    """
+    Show plugin details from index.
+    """
+    name = name.lower()
+    for plugin in indexes.iter_cache_entries():
+        if plugin.name == name:
+            fmt.echo(
+                f"""Name: {plugin.name}
+Source: {plugin.src}
+Status: {plugin_status(name)}
+Author: {plugin.author}
+Maintainer: {plugin.maintainer}
+Homepage: {plugin.url}
+Index: {plugin.index}
+Description: {plugin.description}"""
+            )
+            return
+    raise exceptions.TutorError(f"Plugin could not be found: {name}")
+
+
+def plugin_status(name: str) -> str:
+    """
+    Return the status of a plugin. Either: "enabled", "installed" or "not installed".
+    """
+    if plugins.is_loaded(name):
+        return "enabled"
+    if plugins.is_installed(name):
+        return "installed"
+    return "not installed"
+
+
+@click.group(name="index", short_help="Manage plugin indexes")
+def index_command() -> None:
+    """
+    Manage plugin indices.
+
+    A plugin index is a list of Tutor plugins. An index can be public and shared with
+    the community, or private, for instance to share plugins with a select group of
+    users. Plugin indexes are a great way to share your plugins with other Tutor users.
+    By default, only the official plugin index is enabled.
+
+    Plugin indexes are fetched by running:
+
+        tutor plugins update
+
+    Plugin index cache is stored locally in the following subdirectory of the Tutor project environment:
+
+        plugins/index/cache.yml
+    """
+
+
+@click.command(name="list", help="List plugin indices")
```

Comment:
```suggestion
@click.command(name="list", help="List plugin indexes")
```
Both words are correct, but you're using "indexes" everywhere else so I recommend sticking to that.

Reply:
Good catch! I changed the terminology after I wrote the TEP.